test/modules/CloudToDeviceMessageTester/CloudToDeviceMessageReceiver.cs

```diff
@@ -46,14 +64,18 @@ public async Task ReportTestResult(Message message)
             await ModuleUtil.ReportTestResultAsync(this.testResultReportingClient, this.logger, testResultReceived);
         }
 
-        public override async Task StartAsync(CancellationToken ct)
+        public async Task StartAsync(CancellationToken ct)
         {
+            IEnumerable<X509Certificate2> certs = await CertificateHelper.GetTrustBundleFromEdgelet(new Uri(this.workloadUri), this.apiVersion, this.workloadClientApiVersion, this.moduleId, this.moduleGenerationId);
+            OsPlatform.Current.InstallCaCertificates(certs, ((Protocol)Enum.Parse(typeof(Protocol), this.transportType.ToString())).ToTransportSettings());
             Microsoft.Azure.Devices.RegistryManager registryManager = null;
             try
             {
                 registryManager = Microsoft.Azure.Devices.RegistryManager.CreateFromConnectionString(this.iotHubConnectionString);
-                await registryManager.AddDeviceAsync(new Microsoft.Azure.Devices.Device(this.deviceId + "_leaf"), ct);
-                this.deviceClient = DeviceClient.CreateFromConnectionString(this.iotHubConnectionString + ";" + this.gatewayHostName, this.deviceId, Microsoft.Azure.Devices.Client.TransportType.Amqp_Tcp_Only);
+                Microsoft.Azure.Devices.Device device = await registryManager.AddDeviceAsync(new Microsoft.Azure.Devices.Device(this.deviceId), ct);
+                var csb = IotHubConnectionStringBuilder.Create($"{this.iotHubConnectionString};DeviceId={this.deviceId}");
```

Comment:
nit:specify type
```suggestion
                IotHubConnectionStringBuilder csb = IotHubConnectionStringBuilder.Create($"{this.iotHubConnectionString};DeviceId={this.deviceId}");
```

Reply:
do you need to pass ";DeviceId={this.deviceId}"?

Reply:
I tried it without it, and it threw exception. 

I agree it looks odd, though... maybe there is another way, but I don't know it.